addons/im_livechat/models/mail_channel_partner.py

```diff
@@ -12,12 +12,11 @@ def _gc_unpin_livechat_sessions(self):
         """ Unpin livechat sessions with no activity for at least one day to
             clean the operator's interface """
         self.env.cr.execute("""
-            UPDATE mail_channel_partner
-            SET is_pinned = false
-            WHERE id in (
-                SELECT cp.id FROM mail_channel_partner cp
-                INNER JOIN mail_channel c on c.id = cp.channel_id
-                WHERE c.channel_type = 'livechat' AND cp.is_pinned is true AND
-                    cp.write_date < current_timestamp - interval '1 day'
-            )
+            SELECT cp.id FROM mail_channel_partner cp
+            INNER JOIN mail_channel c on c.id = cp.channel_id
+            WHERE c.channel_type = 'livechat' AND cp.is_pinned is true AND
+                cp.last_seen_dt < current_timestamp - interval '1 day'
         """)
+        cp_records = self.env['mail.channel.partner'].browse([id[0] for id in self.env.cr.fetchall()])
+        sessions_to_be_unpinned = cp_records.filtered(lambda cp: cp.message_unread_counter == 0)
+        sessions_to_be_unpinned.write({'is_pinned': False})
```

Comment:
Now that this is done in python, it could be done through the `channel_pin` method so that if the operator still has the page open, the channel will actually be removed without having to refresh the page.

Reply:
We could use `with_user` if the proper user is necessary by the method.
Or do something equivalent to `_execute_channel_pin` directly here.

The important point of the comment is just to have the update sent on the bus, but it would be nice if the search was still done in batch.